app/src/main/java/ch/hikemate/app/utils/PermissionUtils.kt

```diff
@@ -0,0 +1,52 @@
+package ch.hikemate.app.utils
+
+import android.content.Context
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.MultiplePermissionsState
+
+/**
+ * Utility class for handling permissions.
+ *
+ * Strongly inspired (if not copy-pasted) from this Stack Overflow answer:
+ * https://stackoverflow.com/a/40639430
+ */
+object PermissionUtils {
```

Comment:
Comment: This seems to work pretty well and I think it's good you thought about the case whenever you ask already the permission once.

Reply:
Yup, it's actually super important in Android permission management, and it took make hours to figure out lol